manager/orchestrator/task_test.go

```diff
@@ -105,3 +109,56 @@ func TestIsTaskDirtyPlacementConstraintsOnly(t *testing.T) {
 	service.Spec.Task.Placement.Constraints = nil
 	assert.False(t, IsTaskDirtyPlacementConstraintsOnly(service.Spec.Task, task))
 }
+
+// Test Task sorting, which is currently based on
+// Status.AppliedAt, and then on Status.Timestamp.
+func TestTaskSort(t *testing.T) {
+	var tasks []*api.Task
+	size := 5
+	seconds := int64(size)
+	for i := 0; i < size; i++ {
+		time.Sleep(1)
+		task := &api.Task{
+			ID: "id_" + strconv.Itoa(i),
+			Spec: api.TaskSpec{
+				Runtime: &api.TaskSpec_Container{
+					Container: &api.ContainerSpec{
+						Image: "v:1",
+					},
+				},
+			},
+			Status: api.TaskStatus{
+				Timestamp: &google_protobuf.Timestamp{},
+			},
+		}
+
+		task.Status.Timestamp.Seconds = seconds
+		seconds--
+		tasks = append(tasks, task)
+	}
+
+	sort.Sort(TasksByTimestamp(tasks))
+	expected := int64(1)
+	for _, task := range tasks {
+		timestamp := &google_protobuf.Timestamp{}
```

Comment:
same here; this can be

```
timestamp := &google_protobuf.Timestamp{Seconds: expected}
```

This would even be written as;

```go
sort.Sort(TasksByTimestamp(tasks))
for i, task := range tasks {
	expected := &google_protobuf.Timestamp{Seconds: int64(i + 1)}
	assert.Equal(t, expected, task.Status.Timestamp)
}
```

Also wondering if we want to compare by the task's ID as well (or instead), e.g.;

```go
assert.Equal(t, "id_"+strconv.Itoa(size-(i+1)), task.ID)
```